File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/ui/DetailActivity.java
package com.example.sampleapplicationforgm.ui;

import android.os.Bundle;

import com.example.sampleapplicationforgm.R;
import com.example.sampleapplicationforgm.viewmodel.DetailActivityViewModel;

import javax.inject.Inject;

import androidx.appcompat.app.AppCompatActivity;
import dagger.android.AndroidInjection;

public class DetailActivity extends AppCompatActivity {

    @Inject
    DetailActivityViewModel detailActivityViewModel;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        AndroidInjection.inject(this);
        super.onCreate(savedInstanceState);
        ActivityDetailBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_detail);
        binding.setDetailActivityViewModel(detailActivityViewModel);
    }
}

File: RepositoryPattern/SampleApplicationForGM/app/src/main/java/com/example/sampleapplicationforgm/ui/DetailActivity.kt
package com.example.sampleapplicationforgm.ui

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.example.sampleapplicationforgm.R
import com.example.sampleapplicationforgm.databinding.ActivityDetailBinding
import com.example.sampleapplicationforgm.viewmodel.DetailActivityViewModel
import dagger.android.AndroidInjection
import javax.inject.Inject

class DetailActivity : AppCompatActivity() {
    @JvmField
    @Inject
    var detailActivityViewModel: DetailActivityViewModel? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        AndroidInjection.inject(this)
        super.onCreate(savedInstanceState)
        val binding: ActivityDetailBinding = DataBindingUtil.setContentView(this, R.layout.activity_detail)
        binding.detailActivityViewModel = detailActivityViewModel
    }
}